src/content/Angular-CSharp/ClientApp/proxy.conf.js

```diff
@@ -17,7 +17,10 @@ const PROXY_CONFIG = [
 //#endif
    ],
     target: target,
-    secure: false
+    secure: false,
+    headers: {
+      Connection: 'Keep-Alive'
```

Comment:
This should only make a difference to the server if HTTP/1.0 (or HTTP/0.9) requests are proxied. Or if this is overriding an `Connection: close` header. Do we know what's actually happening? Do we have a network trace of the failed proxy attempt? @Tratcher 

And nit: `keep-alive` is normally lower case when it's part of the `Connection` header, but it is case-insensitive, so it doesn't really matter.

Reply:
@halter73 I don't have insights into the JS frontend proxy, it's only an issue on Mac and Linux, the most I can tell you is that the change solves the issue and that is what was recommended in several GH issues when I searched for the error.

Reply:
Will this ever proxy WebSocket requests? If so, this might break it. That's why we use `proxy_set_header Connection $connection_upgrade;` instead of `proxy_set_header Connection keep-alive;` for [our nginx.conf in our SignalR docs](https://docs.microsoft.com/en-us/aspnet/core/signalr/scale?view=aspnetcore-6.0#linux-with-nginx).

The Connection header also shouldn't be sent for HTTP/2 requests. [Kestrel will abort the entire connection](https://github.com/dotnet/aspnetcore/blob/main/src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs#L1349-L1352) if it sees an `Connection` header in an HTTP/2 request. I'm not sure if the proxy would filter that out for HTTP/2.

Reply:
I'm concerned we keep adding these workarounds without understanding them. This header shouldn't be necessary for most scenarios.

Reply:
@Tratcher can you help troubleshoot what's going on here? 

This code path is only used during development, and I can't tell whether the issue is from kestrel or from the http-proxy-middleware npm package.

For what it matters to us in the templates, this is only a development experience that doesn't affect production code, so I'm way more concerned on fixing the issue than getting to the bottom of the problem. Specially if its not something on Kestrel.

Reply:
If you want to try it for yourself, according to CTI, this problem reproduces on Linux

Reply:
Kestrel Trace level server logs with ConnectionLogging would help answer that.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-6.0#connection-logging

Reply:
@halter73 @Tratcher You have a trace on your email.

With that said, I'm going to move ahead with this fix.

Reply:
Discussed offline, the proxy is sending Connection: close, so we need to set this to avoid that.

Reply:
It's unfortunate that the proxy tries to reuse the connection after sending a `Connection: close` header, but I guess that's out of our control. Could we only apply this when the `Connection: close` request header is present using [onProxyReq](https://github.com/chimurai/http-proxy-middleware#intercept-and-manipulate-requests)?